pkg/test-fw/ig/ig.go

```
@@ -0,0 +1,273 @@
+		// do not return error, it is what we were expecting.
+		var exiterr *exec.ExitError
+		if ok := errors.As(err, &exiterr); !ok {
+			return err
+		}
+
+		waitStatus, ok := exiterr.Sys().(syscall.WaitStatus)
+		if !ok {
+			return err
+		}
+
+		if waitStatus.Signal() != sig {
+			return err
+		}
+
+		return nil
+	}
+
+	return err
+}
+
+func (ig *IG) stop() error {
+	if !ig.started {
+		return fmt.Errorf("Warn(%v): trying to stop command but it was not started", ig.cmd)
+	}
+
+	if err := ig.kill(); err != nil {
+		return err
+	}
+
+	ig.started = false
+	return nil
+}
+
+type option func(*IG)
+
+func WithPath(path string) option {
+	return func(ig *IG) {
+		ig.path = path
+	}
+}
+
+func WithImage(image string) option {
+	return func(ig *IG) {
+		ig.image = image
+	}
+}
+
+// Flags should be in the form: "--flag_name=value" or "-shorthand=value"
+func WithFlags(flags ...string) option {
+	return func(ig *IG) {
+		ig.flags = flags
+	}
+}
+
+// Elements of env should be in form: "ENV_VARIABLE_NAME=value"
+func WithEnv(env ...string) option {
+	return func(ig *IG) {
+		ig.env = env
+	}
+}
+
+func WithStartAndStop() option {
+	return func(ig *IG) {
+		ig.StartAndStop = true
+	}
+}
+
+// Runs "ig version" to get the version string
+func getIgVersionString(path string) (string, error) {
+	cmd := exec.Command(path, "version")
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+	return out.String(), nil
+}
+
+func extractIgVersion(str string) (semver.Version, error) {
+	parsedVersion, err := semver.ParseTolerant(str)
+	if err != nil {
+		return parsedVersion, fmt.Errorf("parsing version from string[%s]: %w]", str, err)
+	}
+	return parsedVersion, nil
+}
+
+// New creates a new IG configured with the options passed as parameters.
+// Supported parameters are:
+//
+//	WithImage(gadget_image)
+//	WithPath(string)
+//	WithEnv(...string)
+//	WithFlags(...string)
+//	WithStartAndStop()
+func New(opts ...option) (*IG, error) {
```

As this is only designed for testing purposes, I think this method should take a testing.T object and call t.Fail (or similar) instead of returning errors. (This will remove a lot of boilerplate error handling code around).